library/md4.c

```diff
@@ -377,6 +410,12 @@ int mbedtls_md4_self_test( int verbose )
         mbedtls_printf( "\n" );
 
     return( 0 );
+
+fail:
+    if( verbose != 0 )
+        mbedtls_printf( "failed\n" );
+
+    return( 1 );
 }
```

Comment:
In case `mbedtls_md4_ext` fails, we should return its error code here instead of `1`.